app/code/Magento/Msrp/Pricing/MsrpPriceCalculator.php

```diff
@@ -23,9 +23,15 @@ class MsrpPriceCalculator implements MsrpPriceCalculatorInterface
     /**
      * @param array $msrpPriceCalculators
      */
-    public function __construct(array $msrpPriceCalculators)
+    public function __construct(?array $msrpPriceCalculators = null)
     {
-        $this->msrpPriceCalculators = $this->getMsrpPriceCalculators($msrpPriceCalculators);
+        /*
+         * This is a workaround for the case of modules Magento_MsrpConfigurableProduct and Magento_MsrpGroupedProduct
```

Comment:
We shouldn't mention these modules here as well :)

Could you please describe what is the root cause of this issue and then we will try to figure out a proper code change together.

Reply:
@orlangur 

I removed the comment.

Route cause is:

$this->msrpPriceCalculators = $this->getMsrpPriceCalculators($msrpPriceCalculators);

cannot be called if the modules Magento_MsrpConfigurableProduct and Magento_MsrpGroupedProduct are disabled.